From 5934c87a79a10942645e12fea0c99f2c24eae01f Mon Sep 17 00:00:00 2001
From: crusaderky <crusaderky@gmail.com>
Date: Mon, 17 Nov 2025 12:18:09 +0000
Subject: [PATCH] Fix ImportError in conda-forge tests

---
 h5py/tests/test_file.py | 37 +++++++++++++++----------------------
 1 file changed, 15 insertions(+), 22 deletions(-)

diff --git a/h5py/tests/test_file.py b/h5py/tests/test_file.py
index 8d2c9281..90b9b60d 100644
--- a/h5py/tests/test_file.py
+++ b/h5py/tests/test_file.py
@@ -13,13 +13,14 @@
     Tests all aspects of File objects, including their creation.
 """
 
+import multiprocessing
 import os
 import stat
 import pickle
 import tempfile
 import time
-import subprocess
 import sys
+from concurrent.futures import ProcessPoolExecutor
 from hashlib import sha256
 
 import pytest
@@ -1005,31 +1006,23 @@ class TestFileLocking:
         """Test file locking option from different concurrent processes"""
         fname = tmp_path / "test.h5"
 
-        def open_in_subprocess(filename, mode, locking):
-            """Open HDF5 file in a subprocess and return True on success"""
-            h5py_import_dir = str(pathlib.Path(h5py.__file__).parent.parent)
-
-            process = subprocess.run(
-                [
-                    sys.executable,
-                    "-c",
-                    f"""
-import sys
-sys.path.insert(0, {h5py_import_dir!r})
-import h5py
-f = h5py.File({str(filename)!r}, mode={mode!r}, locking={locking})
-                    """,
-                ],
-                capture_output=True)
-            return process.returncode == 0 and not process.stderr
-
         # Create test file
-        with h5py.File(fname, mode="w", locking=True) as f:
+        with File(fname, mode="w", locking=True) as f:
             f["data"] = 1
 
-        with h5py.File(fname, mode="r", locking=False) as f:
+        ctx = multiprocessing.get_context("spawn")
+        with (
+            ProcessPoolExecutor(mp_context=ctx, max_workers=1) as ex,
+            File(fname, mode="r", locking=False) as f,
+        ):
             # Opening in write mode with locking is expected to work
-            assert open_in_subprocess(fname, mode="w", locking=True)
+            ex.submit(open_in_subprocess, fname, mode="w", locking=True).result()
+
+
+def open_in_subprocess(*args, **kwargs):
+    """Open and close HDF5 file in a subprocess"""
+    with File(*args, **kwargs):
+        pass
 
 
 @pytest.mark.skipif(
-- 
2.43.0

