Index: setup_configure.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup_configure.py	(revision f525d1c180f6b4bc59726faf833fd50a26d57d83)
+++ setup_configure.py	(revision 38787d70a31d06db9a41b2afe438a0ff7d15b64b)
@@ -59,7 +59,7 @@
         'hdf5_libdir': os.environ.get('HDF5_LIBDIR'),
         'hdf5_pkgconfig_name': os.environ.get('HDF5_PKGCONFIG_NAME'),
         'hdf5_version': os.environ.get('HDF5_VERSION'),
-        'mpi': os.environ.get('HDF5_MPI') == "ON",
+        'mpi': os.environ.get('HDF5_MPI'),
     }
 
 
