From d36628242c5b9c032b2f0f64bfffc3caeda963ba Mon Sep 17 00:00:00 2001
From: Thomas Kluyver <thomas@kluyver.me.uk>
Date: Thu, 19 Sep 2019 17:23:12 +0200
Subject: [PATCH 1/2] Backport PR #1319: 1244 ppc64le tests

---
 .travis.yml  |  7 +++++++
 h5py/h5t.pyx | 16 +++++++---------
 2 files changed, 14 insertions(+), 9 deletions(-)

diff --git a/.travis.yml b/.travis.yml
index 5563fcf5..a779b8a9 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -123,6 +123,13 @@ matrix:
         HDF5_VERSION=1.10.5
         HDF5_DIR=$HDF5_CACHE_DIR/$HDF5_VERSION
 
+    - python: 3.7
+      env:
+      - TOXENV=py37-test-deps
+        HDF5_VERSION=1.10.5
+        HDF5_DIR=$HDF5_CACHE_DIR/$HDF5_VERSION
+      os: linux-ppc64le
+
     # MPI tests
     # TODO: We should test with newer versions of HDF5
     - python: 2.7
diff --git a/h5py/h5t.pyx b/h5py/h5t.pyx
index 8ddd424b..ba2d5a1b 100644
--- a/h5py/h5t.pyx
+++ b/h5py/h5t.pyx
@@ -1369,27 +1369,25 @@ cdef class TypeEnumID(TypeCompositeID):
 def _get_float_dtype_to_hdf5():
     float_le = {}
     float_be = {}
-    h5_be_list = [IEEE_F16BE, IEEE_F32BE, IEEE_F64BE, IEEE_F128BE,
-                  LDOUBLE_BE]
-    h5_le_list = [IEEE_F16LE, IEEE_F32LE, IEEE_F64LE, IEEE_F128LE]
-    if MACHINE != 'ppc64le':
-        h5_le_list.append(LDOUBLE_LE)
+    h5_be_list = [IEEE_F16BE, IEEE_F32BE, IEEE_F64BE, IEEE_F128BE, LDOUBLE_BE]
+    h5_le_list = [IEEE_F16LE, IEEE_F32LE, IEEE_F64LE, IEEE_F128LE, LDOUBLE_LE]
+
     for ftype_, finfo, size in _available_ftypes:
         nmant, maxexp, minexp = _correct_float_info(ftype_, finfo)
         for h5type in h5_be_list:
             spos, epos, esize, mpos, msize = h5type.get_fields()
             ebias = h5type.get_ebias()
             if (finfo.iexp == esize and nmant == msize and
-                (maxexp - 1) == ebias
-            ):
+                    (maxexp - 1) == ebias):
                 float_be[ftype_] = h5type
+                break # first found matches, related to #1244
         for h5type in h5_le_list:
             spos, epos, esize, mpos, msize = h5type.get_fields()
             ebias = h5type.get_ebias()
             if (finfo.iexp == esize and nmant == msize and
-                (maxexp - 1) == ebias
-            ):
+                    (maxexp - 1) == ebias):
                 float_le[ftype_] = h5type
+                break # first found matches, related to #1244
     if ORDER_NATIVE == H5T_ORDER_LE:
         float_nt = dict(float_le)
     else:

From 8fd6db835e6c9b26492a60c4d18d22c066d1e399 Mon Sep 17 00:00:00 2001
From: Thomas A Caswell <tcaswell@bnl.gov>
Date: Fri, 20 Sep 2019 09:47:30 +0200
Subject: [PATCH 2/2] STY: fix whitespace

---
 AUTHORS | 2 --
 1 file changed, 2 deletions(-)

diff --git a/AUTHORS b/AUTHORS
index aca236ab..f1238180 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -121,5 +121,3 @@ Authors are sorted by number of commits.
 * Niru Maheswaranathan
 * Paco Hope
 * Felix Yan
-
-

